scripts/pr-add-size-label.sh

```diff
@@ -0,0 +1,389 @@
+#!/bin/bash
+
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+readonly description=$(cat <<EOT
+Script to add a "size" label to a PR. The size is
+calculated as the sum of changes (total additions and deletions).
+The size total is then looked up in the size_ranges tables to
+determine the appropriate GitHub label to apply to the PR.
+EOT
+)
+
+readonly script_name=${0##*/}
+
+set -o errexit
+set -o errtrace
+set -o nounset
+set -o pipefail
+
+[ -n "${DEBUG:-}" ] && set -o xtrace
+
+# Since this script relies on the output of various commands,
+# protect against the future possibility of those commands being
+# internationalised.
+export LC_ALL="C"
+export LANG="C"
+
+readonly label_prefix="size/"
+
+# Ranges that are used to determine which sizing label should be
+# applied to a PR.
+#
+# The range value must be in one of the following formats:
+#
+# - "<maximum" (value less than the specified maximum).
+# - "minimum-maximum" (open interval value between the specified bounds).
+# - ">minimum" (value greater than the specified minimum).
+#
+# Notes:
+#
+# - The values specified here are arbitrary and may need tweaking
+#   (or overriding with the appropriate environment variable).
+# - The ranges must not overlap.
+readonly default_tiny_range='<10'
+readonly default_small_range='10-49'
+readonly default_medium_range='50-100'
+readonly default_large_range='101-500'
+readonly default_huge_range='>500'
```

Comment:
Those are fine to start with and we can tweak them accordingly depending on the repo, depending on what we learn after some time using them.